node/node.go

```diff
@@ -1055,7 +1055,7 @@ func startStateSync(ssR *statesync.Reactor, bcR fastSyncReactor, conR *cs.Reacto
 	}
 
 	go func() {
-		state, err := ssR.Sync(stateProvider, config.DiscoveryTime)
+		state, err := ssR.Sync(context.Background(), stateProvider, config.DiscoveryTime)
```

Comment:
I'd make this a TODO

Reply:
https://essential-go.programming-books.io/context-todo-vs-context-background-d5224e27ff724a33a79cb4e03a5eb333#:~:text=Code%20should%20use%20context.,propagated%20correctly%20in%20a%20program.

Based on that I believe we should keep this as `Background`.

Reply:
At some point there should be a top-level context associated with the process, and the sync operation should not outlive it. That context would be passed into `startStateSync` and make it possible to stop/restart that operation. Therefore, this context should be a TODO, because it will be replaced in the future.

Reply:
I don't understand why that would need it to be a `TODO`.

Reply:
At the top level we're running the `Start` method of the node service right? Would we expect to have a context there? I guess what I'm asking is what should be done to go from `TODO` to `Background`

Reply:
On something more tangential, we'd want state sync to stop if a user runs the `Stop` method. I'm not sure that that actually happens